mantidimaging/core/io/filenames.py

```diff
@@ -154,13 +154,13 @@ def from_directory(cls, path: Union[Path, str]) -> FilenameGroup | None:
         if not path.is_dir():
             raise ValueError(f"path is a file: {path}")
 
-        for file in sorted(path.iterdir()):
-            if file.suffix.lstrip(".") not in IMAGE_FORMAT_EXTENSIONS:
-                continue
-            fg = cls.from_file(file)
-            return fg
+        files = (f for f in path.iterdir() if f.name[0] != '.' and f.suffix[1:] in IMAGE_FORMAT_EXTENSIONS)
```

Comment:
We should probably add the class method to a class docstring so when we forget why 
```python
f for f in path.iterdir() if f.name[0] != '.'
```
exists so we don't remove it as trivial code and update tests to not test this case.

Reply:
- At least if i was new dev in the team an no-one else in the team had long standing knowledge of the project, I'd consider removing it if i had not prior context of this issue.